modules/purge_purger_http/src/Form/ConfigurationForm.php

```diff
@@ -33,25 +40,79 @@ public function getFormID() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $form['test'] = [
+
+    $config = $this->config('purge_purger_http.settings');
+
+    $form['http_settings'] = [
+      '#title' => $this->t('HTTP Settings'),
+      '#description' => $this->t('Configure how custom outbound HTTP requests should
+      be formed.'),
+      '#type' => 'details',
+      '#open' => TRUE,
+
+    ];
+    $form['http_settings']['hostname'] = [
+      '#title' => $this->t('Hostname'),
       '#type' => 'textfield',
-      '#title' => t('Test'),
+      '#default_value' => $config->get('hostname'),
+      '#required' => FALSE,
+    ];
+    $form['http_settings']['port'] = [
+      '#title' => $this->t('Port'),
+      '#type' => 'textfield',
+      '#default_value' => $config->get('port'),
+      '#required' => FALSE,
+    ];
+    /*
+     * @todo We should get token support in the future.
+     */
+    $form['http_settings']['path'] = [
+      '#title' => $this->t('Path'),
+      '#type' => 'textfield',
+      '#default_value' => $config->get('path'),
+      '#required' => FALSE,
+    ];
+    /*
+     * @todo Confirm all relevant HTTP requests are covered.
+     * http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
+     */
+    $form['http_settings']['request_method'] = [
+      '#title' => $this->t('Request Method'),
+      '#type' => 'select',
+      '#default_value' => array_search($config->get('request_method'), $this->request_methods),
+      '#options' => $this->request_methods,
       '#required' => FALSE,
     ];
-    return parent::buildForm($form, $form_state);
-  }
 
-  /**
-   * {@inheritdoc}
-   */
-  public function validateForm(array &$form, FormStateInterface $form_state) {
-    $form_state->setError($form['test'], $this->t('Not yet implemented yet.'));
+    /*
+     * @todo Implement repeatable rows with two text fields for HEADER -> VALUE
+     */
+    $form['headers'] = [
+      '#title' => $this->t('Headers'),
+      '#type' => 'details',
+      '#open' => TRUE,
+    ];
+
+    $form['ssl'] = [
+      '#title' => $this->t('SSL'),
+      '#type' => 'details',
+      '#open' => TRUE,
+    ];
+
+    return parent::buildForm($form, $form_state);
   }
 
   /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->config('purge_purger_http.settings')
+      ->set('hostname', $form_state->getValue('hostname'))
+      ->set('port', $form_state->getValue('port'))
+      ->set('path', $form_state->getValue('path'))
+      ->set('request_method', $this->request_methods[$form_state->getValue('request_method')])
+      ->save();
+
     return parent::submitForm($form, $form_state);
   }
 }
```

Comment:
Dang, it was there at some point. Dunno how it ended up not being there on commit. Thanks, fixed.